pkg/reconciler/taskrun/validation.go

```diff
@@ -2,15 +2,35 @@ package taskrun
 
 import (
 	"errors"
+	"fmt"
+	"net"
+	"sort"
+	"strconv"
 	"strings"
 
+	// alias for clarity, otherwise it'll look like aws.ActuallyWeThoughtAboutThisOne()
+	ec2Helpers "github.com/konflux-ci/multi-platform-controller/pkg/aws"
 	tektonapi "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
 	"k8s.io/apimachinery/pkg/util/validation"
 )
 
 var (
 	errInvalidPlatformFormat    = errors.New("platform must be in format 'label/label' where each label follows Kubernetes RFC 1035 label name format")
 	errMissingPlatformParameter = errors.New("PLATFORM parameter not found in TaskRun parameters")
+	errInvalidNumericValue      = errors.New("value must be a valid integer between 1 and 100")
+	errInvalidIPFormat          = errors.New("value must be a valid IP address in dotted decimal notation")
+
+	// IBM resource validation errors
+	errIBMHostSecretEmpty            = errors.New("host secret value cannot be empty")
+	errIBMHostSecretPlatformMismatch = errors.New("host secret key and value must contain matching platform substring")
+
+	maxInstancesValue = 250
+	// Maximum allocation timeout value in seconds (20 minutes)
+	maxAllocationTimeout = 1200
+	// Maximum static host concurrency
+	maxStaticConcurrency = 8
+	// Maximum pool host age in minutes (24 hours)
+	maxPoolHostAge = 1440
```

Comment:
how can we avoid hardcoding these values? If in future we want to change them, we'll need a new release. Also we'll not be able to have different values for different clusters.